test/config/test_configurable.rb

```diff
@@ -602,6 +602,20 @@ class TestConfigurable < ::Test::Unit::TestCase
             assert_equal([], x6a.obj2)
           end
         end
+
+        test 'strict value type' do
+          class InfFloatConfig
```

Comment:
Probably its inpossible because the class name is referred while constructing the class.
I've moved it to `ConfigurableSpec` to respect the way of sibling tests.